pkg/routeagent/controllers/route/route.go

```diff
@@ -530,35 +545,49 @@ func (r *Controller) handleRemovedEndpoint(obj interface{}) {
 			klog.Errorf("Could not convert object tombstone %v to an Endpoint", tombstone.Obj)
 			return
 		}
-		klog.V(4).Infof("Recovered deleted object '%s' from tombstone", object.GetName())
 	}
-	klog.V(4).Infof("Informed of removed endpoint for route controller object: %v", object)
+	klog.V(log.DEBUG).Infof("Informed of removed endpoint: %v", object.String())
 	hostname, err := os.Hostname()
 	if err != nil {
 		klog.Fatalf("Could not retrieve hostname: %v", err)
 	}
 	if object.Spec.Hostname == hostname {
 		r.cleanRoutes()
+		klog.V(log.DEBUG).Infof("Removed routes from host")
 	}
 
 	if object.Spec.ClusterID == r.clusterID {
 		r.gwVxLanMutex.Lock()
 		defer r.gwVxLanMutex.Unlock()
 
-		klog.V(6).Infof("Endpoint matches the cluster ID of this cluster")
-		err := r.vxlanDevice.deleteVxLanIface()
-		if err != nil {
-			klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %v", err)
-			return
+		if object.Spec.Hostname == hostname {
+			klog.Infof("Local Gateway Endpoint (IP %s) removed: deleting vxlan interface",
+				object.Spec.PrivateIP)
+		} else {
+			klog.Infof("Gateway Endpoint (IP %s) for our cluster removed: deleting vxlan interface",
+				object.Spec.PrivateIP)
 		}
-		r.vxlanDevice = nil
-	}
 
-	klog.V(4).Infof("Removed routes from host")
+		if r.vxlanDevice == nil {
+			klog.Errorf("r.vxlanDevice == nil , this should never happen, endpoint creation event missed?.")
+		} else {
+			err := r.vxlanDevice.deleteVxLanIface()
+			if err != nil {
+				klog.Errorf("Failed to delete the the vxlan interface on endpoint removal: %s", err.Error())
+				return
+			}
+			r.vxlanDevice = nil
```

Comment:
This should be right after L574.

Reply:
What if the failure was for other reason? "could not be deleted?" at least it would be retried on next similar event.

Although I'm not convinced on the delete strategy.

May be we should just keep those interfaces and just delete/create (=update) when necessary. But no deletion if it's not going to be re-created in a different form. That would feel safer in general. Probably something we want to try after release.

Reply:
I'll move it inside, since the most probable cause of error would be if it didn't exist...

Reply:
Done.